src/aws.ts

```diff
@@ -504,6 +507,237 @@ export namespace $AWS {
+              const defaultMethod =
+                methodName.charAt(0).toUpperCase() + methodName.slice(1);
+              const defaultIamActions = [
+                `${defaultServicePrefix}:${defaultMethod}`,
+              ];
+
+              return makeIntegration<
+                `$AWS.SDK.${ServiceKeys}`,
+                (input: any) => Promise<any>
+              >({
+                kind: `$AWS.SDK.${serviceName}`,
+                native: {
+                  bind(context, args) {
+                    const iamOptions = args[1] as unknown as IamOptions;
+
+                    if (!isObjectLiteralExpr(iamOptions)) {
+                      throw new SynthError(
+                        ErrorCodes.Expected_an_object_literal,
+                        `Second argument ('iamOptions') into a SDK call must be an object`
+                      );
+                    }
+
+                    context.resource.addToRolePolicy(
+                      new aws_iam.PolicyStatement({
+                        effect: aws_iam.Effect.ALLOW,
+                        actions: iamOptions.iamActions ?? defaultIamActions,
+                        resources: iamOptions.iamResources,
+                        conditions: iamOptions.iamConditions,
+                      })
+                    );
+                  },
+                  preWarm(preWarmContext) {
+                    preWarmContext.getOrInit({
+                      key: `$AWS.SDK.${serviceName}`,
+                      init: (key, props) =>
+                        new AWS[serviceName](
+                          props?.clientConfigRetriever?.(key)
+                        ),
+                    });
+                  },
+                  call(args, preWarmContext) {
+                    const client: any = preWarmContext.getOrInit({
+                      key: `$AWS.SDK.${serviceName}`,
+                      init: (key, props) =>
+                        new AWS[serviceName](
+                          props?.clientConfigRetriever?.(key)
+                        ),
+                    });
+
+                    return client[methodName](args[0]).promise();
+                  },
+                },
+                asl: (call, context) => {
+                  const iamOptions = call.args[1] as unknown as IamOptions;
+
+                  if (!isObjectLiteralExpr(iamOptions)) {
+                    throw new SynthError(
+                      ErrorCodes.Expected_an_object_literal,
+                      `Second argument ('iamOptions') into a SDK call must be an object`
+                    );
+                  }
+
+                  context.role.addToPrincipalPolicy(
+                    new aws_iam.PolicyStatement({
+                      effect: aws_iam.Effect.ALLOW,
+                      actions: iamOptions.iamActions ?? defaultIamActions,
+                      resources: iamOptions.iamResources,
+                      conditions: iamOptions.iamConditions,
```

Comment:
if iamOptions is an `Argument`, getting a hold of the real value would look something like `call.args[1].expr.getProperty()...`

Also, the properties themselves could be any expression

```ts
new StepFunction(stack, 'sfn', async (input.actions) => {
    $AWS.SDK.CloudWatch.describeAlarms({}, { iamAction: input.actions })
})
```

=> 

`ObjectLiteral({ iamAction: PropAccess("actions", Identifier(input)) })`

You are probably looking to enforce that the entire object is a constant, in which you could do.

```ts
// toConstant returns undefined if the value cannot be converted to a compile/synth time constant.
const constantIamArgs = call.args[1]?.expr ? toConstant(call.args[1].expr) : undefined;

if(!constantIamArgs) {
   throw new SynthError(...);
} else if(typeof contantIamArg.constant !== "object") {
   // error
}
```